burr/core/application.py

```diff
@@ -121,6 +146,9 @@ def _run_function(function: Function, state: State, inputs: Dict[str, Any], name
         )
     state_to_use = state.subset(*function.reads)
     function.validate_inputs(inputs)
+    if "__context" in inputs:
+        # potentially need to remap the __context variable
+        inputs = _remap_context_variable(function.run, inputs)
```

Comment:
I don't think remapping the context variable should be aware of the signature of the run function. Who knows how it would be created...

Rather, let's just hardcode the name-mangled version and unmangle them? Or maybe do it upstream?